includes/class-send-list.php

```diff
@@ -0,0 +1,358 @@
+			$error_message->export_to( $this->get_errors() );
+			return $error_message;
+		}
+
+		// If the passed value isn't in the enum, throw an error.
+		$this->errors->remove( 'newspack_newsletters_send_list_invalid_property_value_' . $key );
+		$property = $schema['properties'][ $key ];
+		if ( isset( $property['enum'] ) && ! in_array( $value, $property['enum'], true ) ) {
+			$error_message = new WP_Error( 'newspack_newsletters_send_list_invalid_property_value_' . $key, __( 'Invalid value for property: ', 'newspack-newsletters' ) . $key );
+			$error_message->export_to( $this->get_errors() );
+			return $error_message;
+		}
+
+		// Cast value to the expected type.
+		settype( $value, $property['type'] );
+
+		$this->{ $key } = $value;
+		return $this->get( $key );
+	}
+
+	/**
+	 * Get any errors associated with this Send_List.
+	 *
+	 * @return WP_Error
+	 */
+	public function get_errors() {
+		return $this->errors;
+	}
+
+	/**
+	 * Get the provider for this Send_List.
+	 */
+	public function get_provider() {
+		return $this->get( 'provider' );
+	}
+
+	/**
+	 * Get the ID for this Send_List.
+	 */
+	public function get_id() {
+		return $this->get( 'id' );
+	}
+
+	/**
+	 * Get the parent list's ID for this Send_List.
+	 */
+	public function get_parent_id() {
+		return $this->get( 'parent_id' );
+	}
+
+	/**
+	 * Get the type for this Send_List: list or sublist.
+	 */
+	public function get_type() {
+		return $this->get( 'type' );
+	}
+
+	/**
+	 * Get the entity type for this Send_List.
+	 */
+	public function get_entity_type() {
+		return $this->get( 'entity_type' );
+	}
+
+	/**
+	 * Get the name for this Send_List.
+	 */
+	public function get_name() {
+		return $this->get( 'name' );
+	}
+
+	/**
+	 * Update the name for this Send_List.
+	 *
+	 * @param string $value The new name.
+	 */
+	public function set_name( $value ) {
+		return $this->set( 'name', $value );
+	}
+
+	/**
+	 * Get the local name for this Send_List, if the entity is also a Subscription_List.
+	 */
+	public function get_local_name() {
+		return $this->get( 'local_name' );
+	}
+
+	/**
+	 * Update the local name for this Send_List, if the entity is also a Subscription_List.
+	 *
+	 * @param string $value The new name.
+	 */
+	public function set_local_name( $value ) {
+		return $this->set( 'local_name', $value );
+	}
+
+	/**
+	 * Get the contact count for this send list.
+	 */
+	public function get_count() {
+		return $this->get( 'count' );
+	}
+
+	/**
+	 * Update the contact count for this send list.
+	 *
+	 * @param string $value The new name.
+	 */
+	public function set_count( $value ) {
+		return $this->set( 'count', $value );
+	}
+
+	/**
+	 * Get a manually set or dynamic label for autocomplete inputs.
+	 * If not manually set, generate from entity_type, name, and count and store.
+	 *
+	 * @return string
+	 */
+	public function get_label() {
+		$stored_label = $this->get( 'label' );
+		if ( ! empty( $stored_label ) ) {
+			return $stored_label;
+		}
```

Comment:
Same as in `to_array()`, why bail if the value exists if we want it up to date with dynamic data?